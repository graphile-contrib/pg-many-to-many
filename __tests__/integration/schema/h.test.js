const { getSchemaConfig } = require("../../helpers");
const core = require("./core");

test("prints a schema using the 'h' database schema", async () => {
  const config = await getSchemaConfig("h");
  return core.test(["h"], config)();
});
