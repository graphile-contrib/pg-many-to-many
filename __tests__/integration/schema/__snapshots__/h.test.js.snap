// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints a schema using the 'h' database schema 1`] = `
"""A location in a connection that can be used for resuming pagination."""
scalar Cursor

type Membership implements Node {
  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  nodeId: ID!

  """Reads a single \`Person\` that is related to this \`Membership\`."""
  personByPersonIdAAndPersonIdB: Person
  personIdA: Int!
  personIdB: Int!

  """Reads a single \`Team\` that is related to this \`Membership\`."""
  teamByTeamId: Team
  teamId: Int!
}

"""
A condition to be used against \`Membership\` object types. All fields are tested
for equality and combined with a logical ‘and.’
"""
input MembershipCondition {
  """Checks for equality with the object’s \`personIdA\` field."""
  personIdA: Int

  """Checks for equality with the object’s \`personIdB\` field."""
  personIdB: Int

  """Checks for equality with the object’s \`teamId\` field."""
  teamId: Int
}

"""A connection to a list of \`Membership\` values."""
type MembershipsConnection {
  """
  A list of edges which contains the \`Membership\` and cursor to aid in pagination.
  """
  edges: [MembershipsEdge]!

  """A list of \`Membership\` objects."""
  nodes: [Membership]!

  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """The count of *all* \`Membership\` you could get from the connection."""
  totalCount: Int!
}

"""A \`Membership\` edge in the connection."""
type MembershipsEdge {
  """A cursor for use in pagination."""
  cursor: Cursor

  """The \`Membership\` at the end of the edge."""
  node: Membership
}

"""Methods to use when ordering \`Membership\`."""
enum MembershipsOrderBy {
  NATURAL
  PERSON_ID_A_ASC
  PERSON_ID_A_DESC
  PERSON_ID_B_ASC
  PERSON_ID_B_DESC
  PRIMARY_KEY_ASC
  PRIMARY_KEY_DESC
  TEAM_ID_ASC
  TEAM_ID_DESC
}

"""An object with a globally unique \`ID\`."""
interface Node {
  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  nodeId: ID!
}

"""Information about pagination in a connection."""
type PageInfo {
  """When paginating forwards, the cursor to continue."""
  endCursor: Cursor

  """When paginating forwards, are there more items?"""
  hasNextPage: Boolean!

  """When paginating backwards, are there more items?"""
  hasPreviousPage: Boolean!

  """When paginating backwards, the cursor to continue."""
  startCursor: Cursor
}

"""A connection to a list of \`Person\` values."""
type PeopleConnection {
  """
  A list of edges which contains the \`Person\` and cursor to aid in pagination.
  """
  edges: [PeopleEdge]!

  """A list of \`Person\` objects."""
  nodes: [Person]!

  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """The count of *all* \`Person\` you could get from the connection."""
  totalCount: Int!
}

"""A \`Person\` edge in the connection."""
type PeopleEdge {
  """A cursor for use in pagination."""
  cursor: Cursor

  """The \`Person\` at the end of the edge."""
  node: Person
}

"""Methods to use when ordering \`Person\`."""
enum PeopleOrderBy {
  ID_A_ASC
  ID_A_DESC
  ID_B_ASC
  ID_B_DESC
  NATURAL
  PERSON_NAME_ASC
  PERSON_NAME_DESC
  PRIMARY_KEY_ASC
  PRIMARY_KEY_DESC
}

type Person implements Node {
  idA: Int!
  idB: Int!

  """Reads and enables pagination through a set of \`Membership\`."""
  membershipsByPersonIdAAndPersonIdB(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: MembershipCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Membership\`."""
    orderBy: [MembershipsOrderBy!] = [PRIMARY_KEY_ASC]
  ): MembershipsConnection!

  """Reads and enables pagination through a set of \`Membership\`."""
  membershipsByPersonIdAAndPersonIdBList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: MembershipCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Membership\`."""
    orderBy: [MembershipsOrderBy!]
  ): [Membership!]!

  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  nodeId: ID!
  personName: String!

  """Reads and enables pagination through a set of \`Team\`."""
  teams(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: TeamCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Team\`."""
    orderBy: [TeamsOrderBy!] = [PRIMARY_KEY_ASC]
  ): PersonTeamsManyToManyConnection!

  """Reads and enables pagination through a set of \`Team\`."""
  teamsList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: TeamCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Team\`."""
    orderBy: [TeamsOrderBy!]
  ): [Team!]!
}

"""
A condition to be used against \`Person\` object types. All fields are tested for equality and combined with a logical ‘and.’
"""
input PersonCondition {
  """Checks for equality with the object’s \`idA\` field."""
  idA: Int

  """Checks for equality with the object’s \`idB\` field."""
  idB: Int

  """Checks for equality with the object’s \`personName\` field."""
  personName: String
}

"""A connection to a list of \`Team\` values, with data from \`Membership\`."""
type PersonTeamsManyToManyConnection {
  """
  A list of edges which contains the \`Team\`, info from the \`Membership\`, and the cursor to aid in pagination.
  """
  edges: [PersonTeamsManyToManyEdge!]!

  """A list of \`Team\` objects."""
  nodes: [Team]!

  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """The count of *all* \`Team\` you could get from the connection."""
  totalCount: Int!
}

"""A \`Team\` edge in the connection, with data from \`Membership\`."""
type PersonTeamsManyToManyEdge {
  """A cursor for use in pagination."""
  cursor: Cursor

  """The \`Team\` at the end of the edge."""
  node: Team
}

"""The root query type which gives access points into the data universe."""
type Query implements Node {
  """Reads and enables pagination through a set of \`Membership\`."""
  allMemberships(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: MembershipCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Membership\`."""
    orderBy: [MembershipsOrderBy!] = [PRIMARY_KEY_ASC]
  ): MembershipsConnection

  """Reads a set of \`Membership\`."""
  allMembershipsList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: MembershipCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Membership\`."""
    orderBy: [MembershipsOrderBy!]
  ): [Membership!]

  """Reads and enables pagination through a set of \`Person\`."""
  allPeople(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: PersonCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Person\`."""
    orderBy: [PeopleOrderBy!] = [PRIMARY_KEY_ASC]
  ): PeopleConnection

  """Reads a set of \`Person\`."""
  allPeopleList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: PersonCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Person\`."""
    orderBy: [PeopleOrderBy!]
  ): [Person!]

  """Reads and enables pagination through a set of \`Team\`."""
  allTeams(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: TeamCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Team\`."""
    orderBy: [TeamsOrderBy!] = [PRIMARY_KEY_ASC]
  ): TeamsConnection

  """Reads a set of \`Team\`."""
  allTeamsList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: TeamCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Team\`."""
    orderBy: [TeamsOrderBy!]
  ): [Team!]

  """Reads a single \`Membership\` using its globally unique \`ID\`."""
  membership(
    """
    The globally unique \`ID\` to be used in selecting a single \`Membership\`.
    """
    nodeId: ID!
  ): Membership

  """Get a single \`Membership\`."""
  membershipByPersonIdAAndPersonIdBAndTeamId(personIdA: Int!, personIdB: Int!, teamId: Int!): Membership

  """Fetches an object given its globally unique \`ID\`."""
  node(
    """The globally unique \`ID\`."""
    nodeId: ID!
  ): Node

  """
  The root query type must be a \`Node\` to work well with Relay 1 mutations. This just resolves to \`query\`.
  """
  nodeId: ID!

  """Reads a single \`Person\` using its globally unique \`ID\`."""
  person(
    """The globally unique \`ID\` to be used in selecting a single \`Person\`."""
    nodeId: ID!
  ): Person

  """Get a single \`Person\`."""
  personByIdAAndIdB(idA: Int!, idB: Int!): Person

  """
  Exposes the root query type nested one level down. This is helpful for Relay 1
  which can only query top level fields if they are in a particular form.
  """
  query: Query!

  """Reads a single \`Team\` using its globally unique \`ID\`."""
  team(
    """The globally unique \`ID\` to be used in selecting a single \`Team\`."""
    nodeId: ID!
  ): Team

  """Get a single \`Team\`."""
  teamById(id: Int!): Team
}

type Team implements Node {
  id: Int!

  """Reads and enables pagination through a set of \`Person\`."""
  members(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: PersonCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Person\`."""
    orderBy: [PeopleOrderBy!] = [PRIMARY_KEY_ASC]
  ): TeamMembersManyToManyConnection!

  """Reads and enables pagination through a set of \`Person\`."""
  membersList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: PersonCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Person\`."""
    orderBy: [PeopleOrderBy!]
  ): [Person!]!

  """Reads and enables pagination through a set of \`Membership\`."""
  membershipsByTeamId(
    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: MembershipCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Only read the last \`n\` values of the set."""
    last: Int

    """
    Skip the first \`n\` values from our \`after\` cursor, an alternative to cursor
    based pagination. May not be used with \`last\`.
    """
    offset: Int

    """The method to use when ordering \`Membership\`."""
    orderBy: [MembershipsOrderBy!] = [PRIMARY_KEY_ASC]
  ): MembershipsConnection!

  """Reads and enables pagination through a set of \`Membership\`."""
  membershipsByTeamIdList(
    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: MembershipCondition

    """Only read the first \`n\` values of the set."""
    first: Int

    """Skip the first \`n\` values."""
    offset: Int

    """The method to use when ordering \`Membership\`."""
    orderBy: [MembershipsOrderBy!]
  ): [Membership!]!

  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  nodeId: ID!
  teamName: String!
}

"""
A condition to be used against \`Team\` object types. All fields are tested for equality and combined with a logical ‘and.’
"""
input TeamCondition {
  """Checks for equality with the object’s \`id\` field."""
  id: Int

  """Checks for equality with the object’s \`teamName\` field."""
  teamName: String
}

"""
A connection to a list of \`Person\` values, with data from \`Membership\`.
"""
type TeamMembersManyToManyConnection {
  """
  A list of edges which contains the \`Person\`, info from the \`Membership\`, and the cursor to aid in pagination.
  """
  edges: [TeamMembersManyToManyEdge!]!

  """A list of \`Person\` objects."""
  nodes: [Person]!

  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """The count of *all* \`Person\` you could get from the connection."""
  totalCount: Int!
}

"""A \`Person\` edge in the connection, with data from \`Membership\`."""
type TeamMembersManyToManyEdge {
  """A cursor for use in pagination."""
  cursor: Cursor

  """The \`Person\` at the end of the edge."""
  node: Person
}

"""A connection to a list of \`Team\` values."""
type TeamsConnection {
  """
  A list of edges which contains the \`Team\` and cursor to aid in pagination.
  """
  edges: [TeamsEdge]!

  """A list of \`Team\` objects."""
  nodes: [Team]!

  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """The count of *all* \`Team\` you could get from the connection."""
  totalCount: Int!
}

"""A \`Team\` edge in the connection."""
type TeamsEdge {
  """A cursor for use in pagination."""
  cursor: Cursor

  """The \`Team\` at the end of the edge."""
  node: Team
}

"""Methods to use when ordering \`Team\`."""
enum TeamsOrderBy {
  ID_ASC
  ID_DESC
  NATURAL
  PRIMARY_KEY_ASC
  PRIMARY_KEY_DESC
  TEAM_NAME_ASC
  TEAM_NAME_DESC
}
`;
